Vagrantfile

```diff
@@ -143,24 +211,51 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+    # ###############################
+    # Set OpenStack provider settings
+    config.vm.provider "openstack" do |os, override|
+      # load creds file, which you should really have
+      creds_file_path = [nil, ''].include?(ENV['OPENSTACK_CREDS']) ? OPENSTACK_CRED_FILE : ENV['OPENSTACK_CREDS']
+      
+      # read in all the lines that look like FOO=BAR as a hash
+      creds = File.exist?(creds_file_path = File.expand_path(creds_file_path)) ?
+        Hash[*(File.open(creds_file_path).readlines.map{ |l| l.strip!; l.split('=') }.flatten)] : {}
+      voc = vagrant_openshift_config['openstack']
+
+      override.vm.box = voc["box_name"]
+      override.vm.box_url = voc["box_url"]
+      # Make sure the private key from the key pair is provided
+      override.ssh.private_key_path = creds['OSPrivateKeyPath'] || "~/.ssh/id_rsa"
+      
+      os.endpoint     = ENV['OS_AUTH_URL'] ? "#{ENV['OS_AUTH_URL']}/tokens" : creds['OSEndpoint'] 
+      os.tenant       = ENV['OS_TENANT_NAME'] || creds['OSTenant']
+      os.username     = ENV['OS_USERNAME']    || creds['OSUsername']
+      os.api_key      = ENV['OS_PASSWORD']    || creds['OSAPIKey']
+      os.keypair_name = voc['key_pair']       || creds['OSKeypair']  || "<OSKeypair>" # as stored in Nova
+      os.flavor       = voc['flavor']         || creds['OSFlavor']   || /m1.small/       # Regex or String
+      os.image        = voc['image']          || creds['OSImage']    || /Fedora/         # Regex or String
+      os.ssh_username = user = voc['ssh_user']|| creds['OSSshUser']  || "root"           # login for the VM instance
+      os.server_name  = ENV['OSE_SERVERNAME'] || voc['server_name']  || 'openshift-host' # name for the instance created
+      full_provision(override.vm, user)
+    end if vagrant_openshift_config['openstack']
+
+
+    # #########################
     # Set AWS provider settings
     config.vm.provider :aws do |aws, override|
-      creds_file_path = ENV['AWS_CREDS'].nil? || ENV['AWS_CREDS'] == '' ? "~/.awscred" : ENV['AWS_CREDS']
+      creds_file_path = ENV['AWS_CREDS'].nil? || ENV['AWS_CREDS'] == '' ? AWS_CRED_FILE : ENV['AWS_CREDS']
       if File.exist?(File.expand_path(creds_file_path))
         aws_creds_file = Pathname.new(File.expand_path(creds_file_path))
         aws_creds      = aws_creds_file.exist? ? Hash[*(File.open(aws_creds_file.to_s).readlines.map{ |l| l.strip!
                                                           l.split('=') }.flatten)] : {}
 
-        override.vm.box               = "dummy"
-        override.vm.box_url           = "https://github.com/mitchellh/vagrant-aws/raw/master/dummy.box"
-        override.vm.synced_folder sync_from, sync_to, disabled: true
+        voc = vagrant_openshift_config['aws']
+        override.vm.box               = voc['box_name'] || "aws-dummy-box"
+        override.vm.box_url           = voc['box_url'] || AWS_BOX_URL
+        override.vm.synced_folder sync_from, sync_to, disabled: (voc['sync_code'] ? false: true)
```

Comment:
Makes more sense to do a git sync if everything's set up for it. Would be nice for experience to be consistent between providers though...